### ISSUE TITLE
Fullscreen doesn't work properly with multiple screens

### ISSUE THREAD

**Issue by brunodOut** [rdesktop/rdesktop#396]
Hello,

I have two joint monitors on my desktop, and I'm running wayland.

It seems like rdesktop is calculating full screen geometry based on the total screen size. If I set width to 50% it fits on my main screen, for instance, or if I unplug my HDMI display and put it in full screen it also works fine (even after I plug it again). The size of the window, however, seems to be calculated properly as it fits only one of the two displays. 

The main problem is: if you have two displays plugged at the same time and you go fullscreen, you only see half of the remote screen.

I'm not sure about what info I should provide on this report, so please, if you could tell me what is relevant I would thank.

PS: I am willing to fix the bug myself, as I see the project might not have a maintainer, but I will need a heads up about how to properly diagnose the problem at the code level.

**Comment by brunodOut** [comment]
Update: I have digged a little into the code and found out that rdesktops uses a Screen struct to retrieve dimensions in the moment it switches fullscreen. The dimensions present in that struct, as far as I tested, are the total available size, with the total width of the two monitors summed (at least when they are set to extend the screen, not duplicate).

Researching a little bit, I found out the X extension libXxf86vm has a way to retrieve the actual dimensions in a single monitor, I did some tests, it's not 100% clear to me how, in details, it works, but it does (I'll study it a little bit more) through a struct specific for that. I need to mention, I only tested it on Xwayland, and should do more tests also to see in which conditions it would work.

My question is: does it sound like a good solution? Anyone has any other idea?

Best regards,
Bruno
